### ISSUE TITLE
show spinner centered of the screen

### ISSUE THREAD

**Issue by hmarina** [ebrelsford/Leaflet.loading#25]
I would like to show centered loading spinner position, is there way to show separate and centered of the screen.
thank you.

**Comment by ebrelsford** [comment]
I don't see why not, have you tried changing the CSS to center it?

**Comment by nenadvicentic** [comment]
I just played with this (3 years later). It does not seem to be achievable without hacking into Leaflet.

Leaflet puts each element extending `L.Control` into one of four containers: `topleft`, `topright`, `bottomleft` and `bottomright` which is determined by [optional `position` property](https://leafletjs.com/reference-1.5.0.html#control-position) of a `L.Control`. This containers are set to `position: absolute` and stitched to the egdes of `leaflet-control-container` which has position relative and same dimensions as full map.

In order to position `LoadingControl` to the center of a map, it needs to be placed directly into `leaflet-control-container` and not into child containers. This is currently not possible.

**Comment by ebrelsford** [comment]
Yes, the plugin would likely need to put the loading element somewhere outside of the typical Leaflet control elements if we wanted to do this.

I'm not convinced centering the spinner is a good idea design-wise and generally prefer to stick with the standard Leaflet controls configuration, so I'm not going to work on this. If someone wanted to work on a pull request I might consider merging it.
